filecoin-project/lily
# Generic in memory lily testing via vectors

### What
Revive vector tests - this PR adds the logic for testing lily via test vectors!
#### Add new vectors:
Add a vector to `build/test-vectors/vectors.json`, see `build/test-vectors/README.md` for instruction on how to do this

#### Execute vectors
Right now we only have vectors from the calibration network. Mainnet vectors can be added later with no modification to the test suite. 
When executing `make itest-calibration` for the follow occurs:
1. Vectors are downloaded (based on`vector.json`) to `/var/tmp/lily-test-vectors/<network_name>`. If the vector files are cached their contents is simply validated.
2. Ensure the expected vectors are present on the filesystem, failing the test suite otherwise.
3. Connect to timescale instance, migrating to the latest version if required.
4. Construct a full in-memory lily node, configured to talk to storage from the previous step
5. Execute a walk over each vector, the range of the walk is configured via the `to` and `from` fields in `vectors.json`.
6. Extract state from the vector using lotus api methods.
7. Compare data returned by lotus API with contents of storage, passing if equivalent, failing otherwise.

#### How do I run this?
`make dockerup && make itest-calibnet`
